src/msix/pack/AppxBundleWriter.cpp

```diff
@@ -0,0 +1,240 @@
+//
+//  Copyright (C) 2019 Microsoft.  All rights reserved.
+//  See LICENSE file in the project root for full license information.
+// 
+
+#include "AppxPackaging.hpp"
+#include "AppxBundleWriter.hpp"
+#include "MsixErrors.hpp"
+#include "Exceptions.hpp"
+#include "ContentType.hpp"
+#include "Encoding.hpp"
+#include "ZipObjectWriter.hpp"
+#include "AppxManifestObject.hpp"
+#include "ScopeExit.hpp"
+#include "FileNameValidation.hpp"
+#include "StringHelper.hpp"
+
+#include <string>
+#include <memory>
+#include <future>
+#include <algorithm>
+#include <functional>
+
+#include <zlib.h>
+
+namespace MSIX {
+
+    AppxBundleWriter::AppxBundleWriter(IMsixFactory* factory, const ComPtr<IZipWriter>& zip, std::uint64_t bundleVersion)
+        : m_factory(factory), m_zipWriter(zip)
+    {
+        m_state = WriterState::Open;
+        if(bundleVersion == 0)
+        {
+            SYSTEMTIME time = {0};
+            GetSystemTime(&time);

```

Comment:
This won't work for non windows platforms. You need to either define a PAL or to look for c/c++ std functions that gives you this information

Reply:
Replaced with std::time_t and formatted suing std::put_time